src/algebraic_topology/simplex_category.lean

```diff
@@ -402,4 +402,149 @@ full_subcategory_inclusion _
+begin
+  split,
+  { introsI m x y h,
+    have H : const n x ≫ f = const n y ≫ f,
+    { dsimp, rw h },
+    change (n.const x).to_preorder_hom 0 = (n.const y).to_preorder_hom 0,
+    rw cancel_mono f at H,
+    rw H },
+    { intro hyp_inj,
+      exact concrete_category.mono_of_injective _ hyp_inj }
+end
+
+/-- A morphism in `simplex_category` is an epimorphism if and only if it is a surjective function
+-/
+lemma epi_iff_surjective {n m : simplex_category} {f: n ⟶ m} :
+  epi f ↔ function.surjective f.to_preorder_hom :=
+begin
+  split,
+  { intro hyp_f_epi,
+    intro x,
+    by_contradiction h_ab,
+    rw not_exists at h_ab,
+    -- The proof is by contradiction: assume f is not surjective,
+    -- then introduce two non-equal auxiliary functions equalizing f, and get a contradiction.
+    -- First we define the two auxiliary functions.
+    set chi_1 : m ⟶ [1] := hom.mk
+    ⟨λ u, if u ≤ x then 0 else 1,
+     by { intros a b a_leq_b,
+          by_cases a ≤ x,
+          { simp [h], obviously },
+          { simp [h],
+            have b_ge_x : ¬(b ≤ x),
+            { by_contra b_leq_x,
+              have a_leq_x : a ≤ x,
+              { transitivity b,
+                exact a_leq_b,
+                exact b_leq_x },
+                exact h a_leq_x },
+          simp [b_ge_x]}}⟩ with def_chi_1,
+    set chi_2 : m ⟶ [1] := hom.mk
+    ⟨λ u, if u < x then 0 else 1,
+     by { intros a b a_leq_b,
+          by_cases (a = b),
+          { rw h },
+          rename h a_neq_b,
+          have a_lt_b : a < b,
+          { cases nat.eq_or_lt_of_le a_leq_b,
+            { exfalso, exact a_neq_b (subtype.eq h) },
+            { exact h }},
+          by_cases a < x,
+          { simp [h], obviously },
+          { simp [h],
+            have b_geq_x : ¬(b < x),
+            { by_contra b_leq_x,
+              have a_lt_x : a < x,
+              { transitivity b,
+                exact a_lt_b,
+                exact b_leq_x },
+                exact h a_lt_x },
+            simp [b_geq_x] }}⟩ with def_chi_2,
+    -- The two auxiliary functions equalize f
+    have f_comp_chi_i : f ≫ chi_1 = f ≫ chi_2,
+    { dsimp,
+      ext,
+      rw [hom.to_preorder_hom_mk, hom.to_preorder_hom_mk, preorder_hom.comp_coe],
+      rw [hom.to_preorder_hom_mk, hom.to_preorder_hom_mk, preorder_hom.comp_coe],
+      rw [function.comp_app, function.comp_app],
+      have f_x_1_ne_x : ¬((f.to_preorder_hom) x_1 = x) := h_ab x_1,
+      by_cases ((f.to_preorder_hom) x_1) ≤ x,
+      { have f_x_1_lt_x : ((f.to_preorder_hom) x_1) < x,
+        { cases nat.eq_or_lt_of_le h,
+          exfalso,
+          replace h_1 := subtype.eq h_1,
+          exact f_x_1_ne_x h_1,
+          exact h_1 },
+        simp [f_x_1_lt_x, h] },
+      { have n_f_x_1_lt_x : ¬(((f.to_preorder_hom) x_1) < x),
+      { erw not_iff_not_of_iff nat.lt_iff_le_not_le,
+        rw not_and,
+        intro h3, exfalso,
+        refine h h3 },
+        simp [n_f_x_1_lt_x, h] }},
```

Comment:
I made some local suggestions first, but it turns out `simp` is quite magical here:
```suggestion
      simp [le_iff_lt_or_eq, h_ab x_1] },
```

Reply:
Wow, this is a magical simp indeed!